docs/reference/islands.md

```diff
@@ -49,30 +53,18 @@ export default function InteractiveComponent({ defaultCount }) {
-
-The available hydration types are:
-
-- `load` (default)
-- `idle`
-- `visible`
-
-These types can be set as the hydrateOn prop on the Island Component, for example:
-
-```javascript
-<Island module={InteractiveComponent} hydrateOn=”visible” />
-```
-
-For islands with the `idle` hydration type we use [requestIdleCallback](https://developer.mozilla.org/en-US/docs/Web/API/Window/requestIdleCallback), allowing the hydration to be deferred. This is good for lower priority components, allowing client resources to be used first on higher priority.
-
-For Islands with the `visible` hydration type we don’t hydrate until the element is visible on screen by using the [Intersection Observer API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API). This mode is good for components that aren't visible to the user immediately, e.g., if they are further down a long page.
-
-Hydrating means downloading and initializing your Island component code, so using these different hydration types strategically to defer some of that work can help boost initial page load performance!
-
-### `clientOnly`
-
-`boolean`
-
-This is set to `false` by default, but when set to `true` then the Island won’t be rendered on the server. This can be useful for components that rely on logic/libraries that can only run in the browser.
+Name | Type | Default | Description
+-- | -- | -- | --
+<span style="color:white">**clientOnly**</span> | `boolean` | `false` | This is set to `false` by default, but when set to `true` then the Island won’t be rendered on the server. This can be useful for components that rely on logic/libraries that can only run in the browser.
+<span style="color:white">**hydrateOn**</span> | `"load" \| "visible" \| "idle"` | `"load"` | When rendering a page with Islands on the server the output includes a script to initialize Islands on the client. Hydrating means downloading and initializing the Island component code, so using these different hydration types strategically to defer some of that work can help boost initial page load performance! See [Hydration Types](#hydration-types) below for more information.
+<span style="color:white">**id**</span> | `string` | - | This value is passed through to the island's `wrapperTag` id HTML attribute.
+<span style="color:white">**module***</span> | `React.Component` | - | This can be any React component as long as the import URL for this component includes the `?island` query parameter. Refer to the `InteractiveComponent` example at the top of the docs.
+<span style="color:white">**Wrapper**</span> | `React.Component` | - | If the default `div` or `wrapperTag` prop is not sufficient, you can pass a custom component to be used as the island's wrapper component.
```

Comment:
About `Wrapper`, some suggestions...
  - I think we should move it to after `wrapperTag` so that folx see `wrapperTag` first (which hopefully is more commonly needed/used)
  - The main reason `Wrapper` is available is to add specific context providers to each island "react tree". Like the [wrappers needed for styled-components, styled-jsx](https://github.hubspot.com/cms-react/reference/styling.html#styled-components), or similar.
  - I'm pretty sure that `Wrapper` cannot be used to customize other attributes on the wrapper than `style` or `className` (because I think that `wrapperTag` is _always_ used and if `Wrapper` is present it will be a child of `wrapperTag`)
  - Not sure we ever tested a `Wrapper` component that actually rendered any DOM elements, I think we only tested with "null" components that only did context stuff and rendered children. I wouldn't be surprised if there were bugs/awkward behavior if `Wrapper` output more than children 😬

Reply:
Gotcha. I will update accordingly. With that being said, is it safe to assume that the below comment found in the styled-components link above is applicable for all usages of the `Wrapper` prop (i.e. requiring the `?client` param)?

<img width="730" alt="Screenshot 2024-05-20 at 3 21 12 PM" src="https://github.com/HubSpot/cms-react/assets/22771842/5462fb37-c29d-4655-8b4d-9a6585c223e2">

Reply:
Yup :+1: